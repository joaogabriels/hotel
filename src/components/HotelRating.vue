<template>
  <div class="row items-start q-gutter-x-xs" :class="{ 'rating-section': hasAmenities }">
    <span>{{ stars }}</span>
    <q-rating :model-value="Number(stars)" size="xs" readonly color="orange-4" class="rating" />
  </div>
</template>

<script setup lang="ts">
defineProps<{
  stars: string;
  hasAmenities?: boolean;
}>();
</script>

<style lang="scss" scoped>
.rating-section {
  border-right: 1px solid rgba(0, 0, 0, 0.12);
  padding-right: 16px;
}

.rating {
  :deep(.q-rating__icon) {
    text-shadow: none;
  }
}
</style>