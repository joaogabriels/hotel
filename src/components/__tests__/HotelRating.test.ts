import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import { installQuasarPlugin } from '@quasar/quasar-app-extension-testing-unit-vitest'

import HotelRating from '../HotelRating.vue'
import { mockHotel } from './mocks'

installQuasarPlugin()

describe('HotelRating', () => {
  it('matches snapshot', () => {
    const wrapper = mount(HotelRating, {
      props: {
        stars: mockHotel.stars,
        hasAmenities: false,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches snapshot with amenities', () => {
    const wrapper = mount(HotelRating, {
      props: {
        stars: mockHotel.stars,
        hasAmenities: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
