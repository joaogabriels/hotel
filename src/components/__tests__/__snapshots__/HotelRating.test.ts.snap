// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HotelRating > matches snapshot 1`] = `
"<div data-v-378063dc=\\"\\" class=\\"row items-start q-gutter-x-xs\\"><span data-v-378063dc=\\"\\">5</span>
  <div data-v-378063dc=\\"\\" class=\\"q-rating row inline items-center q-rating--non-editable text-orange-4 rating\\" style=\\"font-size: 18px;\\" role=\\"radiogroup\\" aria-readonly=\\"true\\">
    <div class=\\"q-rating__icon-container flex flex-center\\" role=\\"radio\\" aria-checked=\\"false\\" aria-label=\\"grade 1\\"><i class=\\"q-icon notranslate material-icons q-rating__icon q-rating__icon--active text-orange-4\\" aria-hidden=\\"true\\" role=\\"presentation\\">grade</i></div>
    <div class=\\"q-rating__icon-container flex flex-center\\" role=\\"radio\\" aria-checked=\\"false\\" aria-label=\\"grade 2\\"><i class=\\"q-icon notranslate material-icons q-rating__icon q-rating__icon--active text-orange-4\\" aria-hidden=\\"true\\" role=\\"presentation\\">grade</i></div>
    <div class=\\"q-rating__icon-container flex flex-center\\" role=\\"radio\\" aria-checked=\\"false\\" aria-label=\\"grade 3\\"><i class=\\"q-icon notranslate material-icons q-rating__icon q-rating__icon--active text-orange-4\\" aria-hidden=\\"true\\" role=\\"presentation\\">grade</i></div>
    <div class=\\"q-rating__icon-container flex flex-center\\" role=\\"radio\\" aria-checked=\\"false\\" aria-label=\\"grade 4\\"><i class=\\"q-icon notranslate material-icons q-rating__icon q-rating__icon--active text-orange-4\\" aria-hidden=\\"true\\" role=\\"presentation\\">grade</i></div>
    <div class=\\"q-rating__icon-container flex flex-center\\" role=\\"radio\\" aria-checked=\\"true\\" aria-label=\\"grade 5\\"><i class=\\"q-icon notranslate material-icons q-rating__icon q-rating__icon--active text-orange-4\\" aria-hidden=\\"true\\" role=\\"presentation\\">grade</i></div>
  </div>
</div>"
`;

exports[`HotelRating > matches snapshot with amenities 1`] = `
"<div data-v-378063dc=\\"\\" class=\\"row items-start q-gutter-x-xs rating-section\\"><span data-v-378063dc=\\"\\">5</span>
  <div data-v-378063dc=\\"\\" class=\\"q-rating row inline items-center q-rating--non-editable text-orange-4 rating\\" style=\\"font-size: 18px;\\" role=\\"radiogroup\\" aria-readonly=\\"true\\">
    <div class=\\"q-rating__icon-container flex flex-center\\" role=\\"radio\\" aria-checked=\\"false\\" aria-label=\\"grade 1\\"><i class=\\"q-icon notranslate material-icons q-rating__icon q-rating__icon--active text-orange-4\\" aria-hidden=\\"true\\" role=\\"presentation\\">grade</i></div>
    <div class=\\"q-rating__icon-container flex flex-center\\" role=\\"radio\\" aria-checked=\\"false\\" aria-label=\\"grade 2\\"><i class=\\"q-icon notranslate material-icons q-rating__icon q-rating__icon--active text-orange-4\\" aria-hidden=\\"true\\" role=\\"presentation\\">grade</i></div>
    <div class=\\"q-rating__icon-container flex flex-center\\" role=\\"radio\\" aria-checked=\\"false\\" aria-label=\\"grade 3\\"><i class=\\"q-icon notranslate material-icons q-rating__icon q-rating__icon--active text-orange-4\\" aria-hidden=\\"true\\" role=\\"presentation\\">grade</i></div>
    <div class=\\"q-rating__icon-container flex flex-center\\" role=\\"radio\\" aria-checked=\\"false\\" aria-label=\\"grade 4\\"><i class=\\"q-icon notranslate material-icons q-rating__icon q-rating__icon--active text-orange-4\\" aria-hidden=\\"true\\" role=\\"presentation\\">grade</i></div>
    <div class=\\"q-rating__icon-container flex flex-center\\" role=\\"radio\\" aria-checked=\\"true\\" aria-label=\\"grade 5\\"><i class=\\"q-icon notranslate material-icons q-rating__icon q-rating__icon--active text-orange-4\\" aria-hidden=\\"true\\" role=\\"presentation\\">grade</i></div>
  </div>
</div>"
`;
